roles/pulp_common/handlers/restart_all_pulp.yml

```diff
@@ -14,6 +14,14 @@
     daemon_reload: true
   become: true
 
+- name: Get services
+  service_facts:
+
 - name: Restart all pulpcore* services
-  command: systemctl restart pulpcore*
+  systemd:
+    daemon_reload: true
+    name: "{{ item.value.name }}"
+    state: restarted
+  when: "item.value.name != 'pulpcore-worker@.service'"
+  with_dict: '{{ ansible_facts.services|dict2items|selectattr("key", "contains", "pulp")|list|items2dict }}'
   become: true
```

Comment:
A while ago, I wrote up a better idea for how to solve this problem. It would take more time to implement though, partially because you need to figure out the correct/best syntax:

https://pulp.plan.io/issues/7006

Reply:
#7006 is pretty cool!